ponys.js

```diff
@@ -15,18 +15,16 @@ export default class {
 			template = templateElement;
 		}
 		template = template.content;
-		url = new URL(url, location.href);
+		url = new URL(url, location.href.startsWith('about:') ? document.baseURI : location.href);
 
 		let script = template.querySelector('script[setup]') || template.querySelector('script');
-
-		return import(
-			'data:text/javascript;base64,' + btoa(
-				script?.text?.replace(
-					/(import|from)\s*("|')(\.{0,2}\/.*?[^\\])\2/g,  // relative imports
-					(match, keyword, quote, path) => keyword + quote + new URL(path, url) + quote
-				)
-			)
-		).then(module => {
+		let moduleScript = script?.text?.replace(
+			/(import|from)\s*("|')(\.{0,2}\/.*?[^\\])\2/g,  // relative imports
+			(match, keyword, quote, path) => keyword + quote + new URL(path, url) + quote
+		);
+		let blobUrl = URL.createObjectURL(new Blob([moduleScript], { type: 'text/javascript' }));
+		
+		return import(`${blobUrl}#tag=${encodeURIComponent(name)}`).then(module => {
```

Comment:
What is the advantage of adding the `#tag=` here? I don't yet see how this might help with tracking/caching, given that the imported blob will not live past the end of this promise chain... do you have an example use case in mind? :thinking:

Reply:
So lets say you're debugging this script, you wont really see the context of the component class. You wont see that it was embedded within a specific set of html for a specific web component. If the user hovers over the tab where this dynamic script is loaded, they will see the blob url with some random guid, which doesn't really tell them much. However with this small adjustment, when the user hovers over the tab, they will see the component name this class was registered for. If still not clear what I mean here, I will include a screenshot later to better clarify.

Its really not necessary for any technical reason, it's simply to help the developer to get their bearings while debugging (imagine stepping into the project for the first time ever). I also experimented with appending a script comment that also contained the component name which also gives the dev their bearings, but I wasn't too fond of modifying the users script on-demand, even though it was quite minor.

I also think this is something where we could improve further by importing actual script src tags appropriately, if specified, which solves this pain point as result:

`<script type="module" src="component1.js" setup></script>`

Reply:
Example visual:

<img width="763" height="535" alt="image" src="https://github.com/user-attachments/assets/40c932e2-151b-4712-b854-e42536f2643e" />

Note: see how you can even do a CTRL+P search in Chrome dev tools and lookup the component by name, which is grea!

Reply:
OK, yeah, that is quite neat... I agree, this is probably worth keeping! Good idea! :blush:

Reply:
I would be content with:
```suggestion
		return import(`${blobUrl}#${name}`).then(module => {
```
here... I figure that the `tag=` is a little superfluous, and carries no syntactic meaning in a fragment identifier, unlike a query string... likewise, I don't think we need to worry about the pre-emptive encoding on this occasion, as the `import()` will do that for us before sending the request.